airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/JobErrorReportingClient.java

```diff
@@ -16,6 +17,9 @@ public interface JobErrorReportingClient {
   /**
    * Report a job failure reason
    */
-  void reportJobFailureReason(StandardWorkspace workspace, final FailureReason reason, final String dockerImage, Map<String, String> metadata);
+  void reportJobFailureReason(@Nullable StandardWorkspace workspace,
```

Comment:
why is this nullable?

Reply:
We talked about this during sprint demos, but adding here for context:

Because connector specs are cached, they don't really run when setting up a connector for any particular workspace. Instead, this job runs when updating connector versions or adding a new connector where there is no related workspace.